qiskit/pulse/macros.py

```diff
@@ -81,11 +81,10 @@ def measure(qubits: List[int],
                     mem_slot = channels.MemorySlot(qubit_mem_slots[inst.channel.index])
                 else:
                     mem_slot = channels.MemorySlot(unused_mem_slots.pop())
-                if inst.channel.index in qubits:
-                    schedule = schedule.insert(time, instructions.Acquire(
-                        inst.duration, inst.channel, mem_slot=mem_slot))
+                schedule = schedule.insert(time, instructions.Acquire(
+                    inst.duration, inst.channel, mem_slot=mem_slot))
             # Measurement pulses should only be added if its qubit was measured by the user
-            elif inst.channels[0].index in qubits:
+            else:
                 schedule = schedule.insert(time, inst)
```

Comment:
```suggestion
            schedule = schedule.insert(time, inst)
```
We can update the inst, and then this is always needed

Reply:
Fixed in https://github.com/Qiskit/qiskit-terra/pull/5971/commits/1b9ee38a78641265ec75dd672b2101a191aa178f